sdk/core/core/src/az_json_read.c

```diff
@@ -0,0 +1,376 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Have you considered placing the JSON-specific code in its own directory, e.g. `sdk/core/json/src` or something similar?  Maybe I've been programming in other languages for too long but having everything for core in the same directory seems cluttered.